util/conversion/conversion.go

```diff
@@ -170,8 +170,11 @@ func FuzzTestFunc(input FuzzTestFuncInput) func(*testing.T) {
 				g.Expect(spokeAfter.ConvertFrom(hubCopy)).To(gomega.Succeed())
 
 				// Remove data annotation eventually added by ConvertFrom for avoiding data loss in hub-spoke-hub round trips
-				metaAfter := spokeAfter.(metav1.Object)
-				delete(metaAfter.GetAnnotations(), DataAnnotation)
+				// NOTE: Kubeadm types does not have ObjectMeta, so this step is executed only when required.
+				metaAfter, ok := spokeAfter.(metav1.Object)
+				if ok {
+					delete(metaAfter.GetAnnotations(), DataAnnotation)
+				}
```

Comment:
If there is not ObjectMeta in the kubeadm types, why use this function to test? Could we pass the encapsulating object instead?

Reply:
With the new conversions, kubeadm v1beta1 and v1beta2 two types are defined as conversion spoke, and the don't have encapsulating objects anymore.

So it was require to introduce this exception given that v1beta1 and v1beta2 (which are mirrored from kubeadm) do not have ObjectMeta.

Reply:
Wouldn't it be better to just use the super-types instead each inner type? Without this logic, we have no way to preserve conversion between types

Reply:
@vincepri I have added a commit where I'm trying to make skipping this operation something that should be required explicitly (vs trying to assume intent)